### PR TITLE
WT-12934 Address remaining issues with zero length files and git diffs in the Code Change Report

### PR DESCRIPTION
**Pull request by jeremythorp**
[WT-12868](https://jira.mongodb.org/browse/WT-12868) created a workaround for newly added 0-length files, however @bzm-jas 's PR for [WT-12837](https://jira.mongodb.org/browse/WT-12837) also has some deleted 0-length files. Each of these files can generate either a deleted file record in the diff, or it may generate a record moving that file to another location as it sees a match in the (0-length) content.

The solution is to also exclude from the diff output file:

- Any deleted files
- Any moved files where the record includes the path of a 0-length file.

### CODE REVIEW

**Review by ajmorton — APPROVED** [review]
LGTM

**Review by bzm-jas — APPROVED** [review]
LGTM! Thanks Jeremy!:)

File: test/evergreen/code_change_report/git_diff_tool.py [hunk 1]
```diff
@@ -60,24 +65,55 @@ def find_zero_length_files(directory: str) -> List[str]:
...
+    working_directory.pop()
+    return moved_zero_length_files
+
+
 def create_diff_file(git_working_tree_dir: str, diff_file: str, verbose: bool) -> None:
     # Exclude files with 0-length from the diff, as pygit2 doesn't handle such diffs well
-    zero_length_files = find_zero_length_files(git_working_tree_dir)
-    exclude_files_param = ' '.join([f"':(exclude){file}'" for file in zero_length_files])
+    zero_length_files = find_zero_length_files(directory=git_working_tree_dir)
+    deleted_files = find_deleted_files(git_working_tree_dir=git_working_tree_dir)
```
[the remark this answers is not in the document]
Comment (jeremythorp) [reply test/evergreen/code_change_report/git_diff_tool.py:97]: That's correct, as we can't tell how big they are (easily), and we don't want/need them anyway.

File: test/evergreen/code_change_report/git_diff_tool.py [hunk 2]
```diff
@@ -52,6 +52,11 @@ def run_command(directory: str, command: str) -> str:
     return output.decode()
 
 
+def get_merge_base_commit(git_working_tree_dir: str) -> str:
+    merge_base_commit = run_command(git_working_tree_dir, "git merge-base develop HEAD").strip()
```
[the remark this answers is not in the document]
Comment (jeremythorp) [reply test/evergreen/code_change_report/git_diff_tool.py:56]: I haven't noticed this so far, but I'll run some tests to check.

File: test/evergreen/code_change_report/git_diff_tool.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremythorp) [reply test/evergreen/code_change_report/git_diff_tool.py]: I've renamed that variable and added extra comments to make the code clearer.

**Pull request merged**